src/components/navigation/Navigation.tsx

```diff
@@ -21,7 +21,7 @@ export interface INavigationChildrenProps {
     onPageClick?: (pageNb: number) => void;
     perPageLabel?: string;
     perPageNumbers?: number[];
-    onPerPageClick?: () => void;
+    onPerPageClick?: (newPerPage: number, currentPerPage: number, currentPage: number) => void;

```

Comment:
I think the parameters should be optional. The user might not need them and might send a function without these parameters.

Reply:
I think you're confused, the onPerPageClick is optional, but here we define what we send to the function. If the "user" don't want to use them they can simply omit them in the receiver

[Example](http://www.typescriptlang.org/play/#src=interface%20SomeProps%20%7B%0D%0A%20%20%20%20f%3F%3A%20(a%3A%20string%2C%20b%3A%20number)%20%3D%3E%20any%3B%0D%0A%7D%0D%0A%0D%0Aconst%20One%3A%20SomeProps%20%3D%20%7B%7D%3B%0D%0A%0D%0Aconst%20Two%3A%20SomeProps%20%3D%20%7B%0D%0A%20%20%20%20f%3A%20()%20%3D%3E%20null%2C%0D%0A%7D%3B%0D%0A%0D%0Aconst%20Three%3A%20SomeProps%20%3D%20%7B%0D%0A%20%20%20%20f%3A%20(a%3A%20string)%20%3D%3E%20a%2C%0D%0A%7D%0D%0A%0D%0Aconst%20Four%3A%20SomeProps%20%3D%20%7B%0D%0A%20%20%20%20f%3A%20(a%3A%20string%2C%20b%3A%20number)%20%3D%3E%20b%2C%0D%0A%7D)

Reply:
I thought the same as @sroy3 but it seems you're right according to your example

Reply:
After looking at it again, I think that currentPerPage and currentPage are optional, and the only required one would be newPerPage. The other arguments are only set if Navigation receives currentPerPage and currentPage as props (in the current implementation).